firmware/src/drivers/pwm.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * CATS Flight Software
+ * Copyright (C) 2023 Control and Telemetry Systems
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "target.h"
+
+namespace driver {
+
+class Pwm {
+ public:
+  Pwm(TIM_HandleTypeDef& timer, uint32_t channel)
+      : m_timer(timer), m_channel(channel), m_period(0U), m_duty_cycle_ticks(0U), m_started(false), m_depth(10000U) {}
+
+  /** Set the frequency of the pwm generator
+   *
+   * @param frequency the frequency in Hz
+   * @return true on success
+   */
+  bool SetFrequency(uint32_t frequency) {
+    // Calculate the prescaler to get as close to the pwm depth as possible
+    uint32_t psc = SystemCoreClock / (m_depth * frequency);
```

Comment:
All of these implementations should be in a .cpp file

https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++-dos-and-donts.md#minimize-code-in-headers